### PR TITLE
Add a test for find_content receipts over uTP

### PR DESCRIPTION
**Pull request by KolbyML**
Testing if uTP tranfers through find_content is implemented correctly

### CODE REVIEW

File: simulators/portal-interop/src/main.rs [hunk 1]
```diff
@@ -480,3 +492,59 @@ dyn_async! {
...
+    async fn test_find_content_receipts_over_utp<'a> (client_a: Client, client_b: Client) {
+        let receipts_key: HistoryContentKey = serde_json::from_value(json!(RECEIPTS_KEY)).unwrap();
+        let receipts_value: HistoryContentValue = serde_json::from_value(json!(RECEIPTS_VALUE)).unwrap();
+
+        let header_with_proof_key: HistoryContentKey = serde_json::from_value(json!(HEADER_WITH_PROOF_KEY)).unwrap();
+        let header_with_proof_value: HistoryContentValue = serde_json::from_value(json!(HEADER_WITH_PROOF_VALUE)).unwrap();
+
+        match client_b.rpc.store(receipts_key.clone(), receipts_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Error storing header with proof for Receipts verification");
```
Comment (kdeme) [inline simulators/portal-interop/src/main.rs:507]: Error storing receipts rather?

File: simulators/portal-interop/src/main.rs [hunk 2]
```diff
@@ -480,3 +492,59 @@ dyn_async! {
...
+            },
+            Err(err) => {
+                panic!("Error storing header with proof for Receipts verification: {err:?}");
```
Comment (kdeme) [inline simulators/portal-interop/src/main.rs:510]: idem^

File: simulators/portal-interop/src/main.rs [hunk 3]
```diff
@@ -480,3 +492,59 @@ dyn_async! {
...
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
+
+        match client_a.rpc.find_content(target_enr, receipts_key.clone()).await {
+            Ok(result) => {
+                match result {
+                    // todo: when clients conform with spec we will need to add a check if
```
Comment (kdeme) [inline simulators/portal-interop/src/main.rs:533]: I think it is fine to add a test that clients fail on, if it is according to spec

File: simulators/portal-interop/src/main.rs (hunk 1 above)
Comment (KolbyML) [reply simulators/portal-interop/src/main.rs]: My bad xD I didn't notice that, resolved

File: simulators/portal-interop/src/main.rs (hunk 2 above)
Comment (KolbyML) [reply simulators/portal-interop/src/main.rs]: I updated it hopefully that is more clear

File: simulators/portal-interop/src/main.rs (hunk 3 above)
Comment (KolbyML) [reply simulators/portal-interop/src/main.rs:533]: It is a field in the json-rpc find_content spec I believe

Comment (kdeme) [reply simulators/portal-interop/src/main.rs:533]: Yeah, that's what I mean. It's already in the spec, so you can activate a test for it. If clients don't support it, it creates somewhat of an incentive to fix it :)

**Review by kdeme — APPROVED** [review]

**Pull request merged**